File: cpp/demos/demo_dwa.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/reactive/dwa.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "dwa", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::DwaPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}

File: python/demos/demo_dwa.py
#!/usr/bin/env python3
"""DWA demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_pure_pursuit.py's
# precedent (the package re-export lives in
# navigation/local_planning/reactive/__init__.py).
from navigation.local_planning.reactive.dwa import Dwa

if __name__ == "__main__":
    run_local("dwa", Dwa)
